js/src/reports/utils.test.js

```diff
@@ -0,0 +1,296 @@
+/**
+ * Internal dependencies
+ */
+import {
+	freeFields,
+	paidFields,
+	MISSING_FREE_LISTINGS_DATA,
+} from '.~/data/utils';
+import {
+	getIdsFromQuery,
+	aggregateIntervals,
+	sumToPerformance,
+	addBaseToPerformance,
+} from './utils';
+
+// Copied from https://github.com/woocommerce/woocommerce-admin/blob/b35156dcf17b44b3a81ea4b4528445b432917fd5/packages/navigation/src/test/index.js#L184-L221
+describe( 'getIdsFromQuery', () => {
+	it( 'if the given query is empty, should return an empty array', () => {
+		expect( getIdsFromQuery( '' ) ).toEqual( [] );
+	} );
+
+	it( 'if the given query is undefined, should return an empty array', () => {
+		expect( getIdsFromQuery( undefined ) ).toEqual( [] );
+	} );
+
+	it( 'if the given query is does not contain any coma-separated numbers, should return an empty array', () => {
+		expect( getIdsFromQuery( 'foo123,bar,baz1.' ) ).toEqual( [] );
+	} );
+
+	describe( 'if the given query contains numbers', () => {
+		it( 'should return an array of them', () => {
+			expect( getIdsFromQuery( '77,8,-1' ) ).toEqual( [ 77, 8, -1 ] );
+		} );
+		it( 'should consider `0` a valid id', () => {
+			expect( getIdsFromQuery( '0' ) ).toEqual( [ 0 ] );
+			expect( getIdsFromQuery( '77,0,1' ) ).toEqual( [ 77, 0, 1 ] );
+		} );
+		it( 'should map floats to integers', () => {
+			expect( getIdsFromQuery( '77,8.54' ) ).toEqual( [ 77, 8 ] );
+		} );
+		it( 'should ignore duplicates', () => {
+			expect( getIdsFromQuery( '77,8,8' ) ).toEqual( [ 77, 8 ] );
+			// Consider two floats that maps to the same integer a duplicate.
+			expect( getIdsFromQuery( '77,8.5,8.4' ) ).toEqual( [ 77, 8 ] );
+		} );
+		it( 'should ignore non numbers entries in the coma-separated list', () => {
+			expect( getIdsFromQuery( '77,,8,foo,null,9' ) ).toEqual( [
+				77,
+				8,
+				9,
+			] );
+		} );
+	} );
+} );
```

Comment:
>  which means both `getIdsFromQuery` in WC Admin and our repo are now the same

I'm afraid not for all cases. 
1. It's still not released, https://github.com/woocommerce/woocommerce-admin/blob/main/packages/navigation/CHANGELOG.md. 
2. We lag quite far with our `/navigation` dependency https://github.com/woocommerce/google-listings-and-ads/blob/trunk/package.json#L63. 
3. What's even worse, we are tied to the version used in WC 5.2 https://github.com/woocommerce/google-listings-and-ads/blob/trunk/google-listings-and-ads.php#L13 what will lag us even longer.

-----

That's said I share the sentiment, of not duplicating the code too much. However, given the weight of technical debt our dependencies have, we will probably need to support this workaround for a quite long time - this would be nice to make sure the workaround works, without any regression.

Reply:
Maybe we could consider removing it from here, and creating a separate file, with a disclaimer/comment like:
```js
/** 
 * This is a test suite for a local copy, of not fully available fix in `@woocommerce/navigation`.
 * If you need to change something here, make sure you do so as well in the `/navigation` package as well.
 */

// Copied from https://github.com/woocommerce/woocommerce-admin/blob/b35156dcf17b44b3a81ea4b4528445b432917fd5/packages/navigation/src/test/index.js#L184-L221
```

Reply:
> 3\. What's even worse, we are tied to the version used in WC 5.2 https://github.com/woocommerce/google-listings-and-ads/blob/trunk/google-listings-and-ads.php#L13 what will lag us even longer.

For more information, the runtime `@woocommerce/navigation` version is bound to the version used in WC by:

https://github.com/woocommerce/google-listings-and-ads/blob/648b1ebd31f0aa2244c5fd02c3bd186909b4b5f9/webpack.config.js#L18-L30

For the longest case, we have to keep it until the minimum support WC version is bump to a version that includes the fix https://github.com/woocommerce/woocommerce-admin/pull/7028

Reply:
That feels unfortunate. Gotcha, thank you all for the explanation. 🙂